addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/FreeboxApiManager.java

```diff
@@ -0,0 +1,572 @@
+        if (StringUtils.isNotEmpty(airPlayPassword)) {
+            request.setPassword(airPlayPassword);
+        }
+        executeUrl("POST", "airmedia/receivers/" + encodeUrl(airPlayName) + "/", gson.toJson(request),
+                EmptyResponse.class, true, false, true);
+    }
+
+    public void reboot() throws FreeboxException {
+        executeUrl("POST", "system/reboot/", null, EmptyResponse.class);
+    }
+
+    private <T extends FreeboxResponse<F>, F> F executeUrl(String httpMethod, String relativeUrl, String requestContent,
+            Class<T> responseClass) throws FreeboxException {
+        return executeUrl(httpMethod, relativeUrl, requestContent, responseClass, true, false, false);
+    }
+
+    private <T extends FreeboxResponse<F>, F> F executeUrl(String httpMethod, String relativeUrl, String requestContent,
+            Class<T> responseClass, boolean retryAuth) throws FreeboxException {
+        return executeUrl(httpMethod, relativeUrl, requestContent, responseClass, retryAuth, false, false);
+    }
+
+    private <T extends FreeboxResponse<F>, F> F executeUrl(String httpMethod, String relativeUrl, String requestContent,
+            Class<T> responseClass, boolean retryAuth, boolean patchTableReponse, boolean doNotLogData)
+            throws FreeboxException {
+        try {
+            Properties headers = null;
+            String token = getSessionToken();
+            if (token != null) {
+                headers = new Properties();
+                headers.setProperty(AUTH_HEADER, token);
+            }
+            InputStream stream = null;
+            String contentType = null;
+            if (requestContent != null) {
+                stream = new ByteArrayInputStream(requestContent.getBytes(StandardCharsets.UTF_8));
+                contentType = HTTP_CALL_CONTENT_TYPE;
+            }
+            logger.debug("executeUrl {} {} requestContent {}", httpMethod, relativeUrl,
+                    doNotLogData ? "***" : requestContent);
+            String jsonResponse = HttpUtil.executeUrl(httpMethod, baseAddress + relativeUrl, headers, stream,
+                    contentType, HTTP_CALL_DEFAULT_TIMEOUT_MS);
+
+            if (patchTableReponse) {
+                // Replace empty result by an empty table result
+                jsonResponse = jsonResponse.replace("\"result\":{}", "\"result\":[]");
+            }
+
+            return evaluateJsonReesponse(jsonResponse, responseClass, doNotLogData);
+        } catch (FreeboxException e) {
+            if (retryAuth && e.isAuthRequired()) {
+                logger.debug("Authentication required: open a new session and retry the request");
+                openSession();
+                return executeUrl(httpMethod, relativeUrl, requestContent, responseClass, false, patchTableReponse,
+                        doNotLogData);
+            }
+            throw e;
+        } catch (IOException e) {
+            throw new FreeboxException(
```

Comment:
Should we rather add the `e` as a cause to the new exception?

Reply:
Done